src/test/scala/mesosphere/marathon/core/task/termination/impl/KillStreamWatcherTest.scala

```diff
@@ -40,4 +44,28 @@ class KillStreamWatcherTest extends AkkaUnitTest {
 
     result.futureValue shouldBe Nil
   }
+
+  "KillStreamWatcher emits already terminated instances" in {
+
+    val empty = MarathonTestHelper.emptyInstance()
+    val unreachableInstance = empty.copy(state = empty.state.copy(condition = Condition.Unreachable))
+
+    val watcher = KillStreamWatcher.watchForKilledInstances(system.eventStream, List(unreachableInstance))
+
+    watcher.runWith(Sink.head).futureValue shouldEqual Done
```

Comment:
Why not test `killedInstanceIds` here?

Reply:
Test of `watchForKilledInstances` already covers `killedInstanceIds`.

Reply:
Yeah but the assertion is not giving much details.

Reply:
`withClue("Watcher should emit instantly emit instances that are already terminal")`?